### PR TITLE
Add method to retrieve original metadata from point cloud files

### PR DESCRIPTION
**Pull request by nyalldawson**
and expose some more useful metadata in layer properties (las version, software, creation date)

eg

image

### CODE REVIEW

File: src/core/pointcloud/qgspointcloudlayer.cpp [hunk 1]
```diff
@@ -422,6 +422,51 @@ QString QgsPointCloudLayer::htmlMetadata() const
                 + ( pointCount < 0 ? tr( "unknown" ) : locale.toString( static_cast<qlonglong>( pointCount ) ) )
                 + QStringLiteral( "</td></tr>\n" );
 
+  const QVariantMap originalMetadata = mDataProvider ? mDataProvider->originalMetadata() : QVariantMap();
+
+  if ( originalMetadata.value( QStringLiteral( "creation_year" ) ).toInt() > 0 && originalMetadata.contains( QStringLiteral( "creation_doy" ) ) )
```
Comment (PeterPetrik) [inline src/core/pointcloud/qgspointcloudlayer.cpp:427]: creation_doy -> creation_day ?

File: tests/src/python/test_qgspointcloudprovider.py [hunk 2]
```diff
@@ -94,6 +94,24 @@ def testMetadataClassStatistics(self):
...
                          3)
 
+    @unittest.skipIf('ept' not in QgsProviderRegistry.instance().providerList(), 'EPT provider not available')
+    def testOriginalMetadataEpt(self):
+        layer = QgsPointCloudLayer(unitTestDataPath() + '/point_clouds/ept/sunshine-coast/ept.json', 'test', 'ept')
+        self.assertEqual(layer.dataProvider().originalMetadata()['major_version'], 1.0)
+        self.assertEqual(layer.dataProvider().originalMetadata()['minor_version'], 2.0)
+        self.assertEqual(layer.dataProvider().originalMetadata()['software_id'], 'PDAL 2.1.0 (Releas)')  #spellok
+        self.assertEqual(layer.dataProvider().originalMetadata()['creation_year'], 2020.0)
+        self.assertEqual(layer.dataProvider().originalMetadata()['creation_doy'], 309.0)
```
Comment (PeterPetrik) [inline tests/src/python/test_qgspointcloudprovider.py:104]: day?

File: src/core/pointcloud/qgspointcloudlayer.cpp (hunk 1 above)
Comment (nyalldawson) [reply src/core/pointcloud/qgspointcloudlayer.cpp:427]: Doy is correct -- it's the "day of year"

File: src/core/pointcloud/qgspointcloudlayer.cpp [hunk 3]
```diff
@@ -422,6 +422,51 @@ QString QgsPointCloudLayer::htmlMetadata() const
...
+                  + QStringLiteral( "</td></tr>\n" );
+  }
+  if ( originalMetadata.contains( QStringLiteral( "major_version" ) ) && originalMetadata.contains( QStringLiteral( "minor_version" ) ) )
+  {
+    myMetadata += QStringLiteral( "<tr><td class=\"highlight\">" )
+                  + tr( "Version" ) + QStringLiteral( "</td><td>" )
+                  + QStringLiteral( "%1.%2" ).arg( originalMetadata.value( QStringLiteral( "major_version" ) ).toString(),
+                      originalMetadata.value( QStringLiteral( "minor_version" ) ).toString() )
+                  + QStringLiteral( "</td></tr>\n" );
+  }
```
Comment (wonder-sk) [inline src/core/pointcloud/qgspointcloudlayer.cpp:444]: For LAS/LAZ files there are few more interesting bits of information:
- dataformat_id - which "Point Data Record Format" is used internally (that also implies which attributes are present)
- scale_x, scale_y, scale_z - what is the precision of coordinates (e.g. 0.01)
- offset_x, offset_y, offset_z - could be interesting, but probably mostly stuff for geeks

Comment (nyalldawson) [reply src/core/pointcloud/qgspointcloudlayer.cpp:444]: Thanks @wonder-sk, I'll add these!

For very useful, common metadata attributes we may want to consider also adding explicit getters to the data provider class, but we can do that when we identify the need for each...

**Pull request merged**